### PR TITLE
Loosen minSdkVersion so this library can be selectively enabled 

### PR DESCRIPTION
**Pull request by hgani**
Hi,

This pull request loosens the minSdkVersion so turbolinks-android can be included and selectively enabled/disabled depending on whether the device has the required SDK version.

Example:
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
      // Implementation using Turbolinks
    }
    else {
      // Non-turbolinks implementation
    }
    
The reason I chose 15 as minSdkVersion is because that's what we use for our production app, so this is the min version that has been tested.

Thanks

### CODE REVIEW

**Comment by lateplate** [comment]
Hi. Sorry, we don't have any plans to support anything below 4.4, which is the lowest version that uses the Chromium-based WebView.

**Comment by hgani** [comment]
Hi @lateplate Sorry please hear me a bit. 

That's not what this pull request is about.

This pull request is for developers who want to support API level below 4.4 for their app, so they have the flexibility of using Turbolinks **only when** their app is running on supported API levels (4.4 and above).

As it is now, turbolinks-android forces all developers to change minSdkVersion in their app. I see that this is why Basecamp 3 is a separate app (not a continuation of Basecamp 2) in AppStore and PlayStore, but not everyone can afford to fork a new app.

Bottom line, this pull request is **not** about supporting lower API levels, it's more about relaxing the restriction so that apps can decide when to use Turbolinks stuff and when not to, which is a common technique in Android development. See "Check System Version at Runtime" at https://developer.android.com/training/basics/supporting-devices/platforms.html

Thanks

**Comment by lateplate** [comment]
I understand what you're saying, but what you're describing is exactly the confusion we want to avoid.

4.4 was the first version of WebView built on Chromium. That's a fundamental assumption from which everything in the library was implemented and tested upon.

We don't want to rely on every developer to implement version checks to implement the library properly. It's much simpler to have a minSdkVersion cutoff and ensuring it works for everyone over 4.4.

You are of course more than welcome to fork the project, change the minSdkVersion, and build it with that one small change for yourself. It still should work great if you implement the version checks.

Incidentally, this has nothing to do with Basecamp 3 or "affording to fork a new app". Again, 4.4 was the first version of WebView based on Chromium, which makes it more predictable to develop against.

**Comment by hgani** [comment]
@lateplate Okay, thanks for clarifying.

Keep up the good work.
